sdk/program/src/log.rs

```diff
@@ -19,12 +19,37 @@ macro_rules! info {
             $arg4 as u64,
             $arg5 as u64,
         )
-    }; // `format!()` is not supported yet, Issue #3099
-       // `format!()` incurs a very large runtime overhead so it should be used with care
-       // ($($arg:tt)*) => ($crate::log::sol_log(&format!($($arg)*)));
+    };
+}
+
+/// Print a message to the log
+///
+/// There are two fast forms:
+/// 1. Single string: `msg!("hi")`
+/// 2. 5 integers: `msg!(1, 2, 3, 4, 5)`
+///
+/// The third form is more generic and incurs a very large runtime overhead so it should be used
+/// with care:
+/// 3. Generalized format string: `msg("Hello: 1, 2, {}", "World", 3)`
+///
+#[macro_export]
+macro_rules! msg {
```

Comment:
Does the implementaiton of `msg` differ from `info`, maybe just have `info` call `msg` to avoid duplication

Reply:
Implementation doesn't differ, not sure there's really much to de-dup though and with duplication I'm 100% sure there's no backwards compatibility issue